packages/@o3r/eslint-plugin/src/rules/typescript/no-multiple-type-configuration-property/no-multiple-type-configuration-property.ts

```diff
@@ -0,0 +1,79 @@
+import { TSESTree } from '@typescript-eslint/experimental-utils';
+import { createRule, isExtendingConfiguration } from '../../utils';
+
+export interface NoMultipleTypeConfigurationPropertyOption {
+  supportedInterfaceNames?: string[];
+}
+
+export default createRule<NoMultipleTypeConfigurationPropertyOption[], 'error' | 'suggestion'>({
+  name: 'no-multiple-type-configuration-property',
+  meta: {
+    hasSuggestions: true,
+    fixable: 'code',
+    type: 'problem',
+    docs: {
+      description: 'Ensures that the configuration property does not accept multiple type.',
+      recommended: 'error'
+    },
+    schema: [
+      {
+        type: 'object',
+        properties: {
+          supportedInterfaceNames: {
```

Comment:
Actually the name is not really correct, as we always check the `NestedConfiguration` and `Configuration`.
Something like `additiionalSupportedInterfaceName` would be more correct :S